performance_regression_gradual_grow_throughput.py

```diff
@@ -182,45 +202,67 @@ def run_gradual_increase_load(self, stress_cmd_templ,
-        if popuplation_distribution == CSPopulateDistribution.GAUSS:
-            return f"GAUSSIAN(1..{self.ROW_COUNT},{self.GAUSS_CENTER},{self.GAUSS_SIGMA})"
-        elif popuplation_distribution == CSPopulateDistribution.UNIFORM:
-            return f"UNIFORM(1..{self.ROW_COUNT})"
-        else:
-            self.log.error("Unsupported cs population distribution")
-            return ""
+        # Keep next 3 lines for debug purpose
+        self.log.debug("es_index: %s", self._test_index)
+        self.log.debug("total_summary: %s", total_summary)
+        self.log.debug("setup_details: %s", setup_details)
+        is_gce = bool(self.params.get('cluster_backend') == 'gce')
+        try:
+            perf_analyzer.check_regression(test_id=self.test_id,
+                                           data=total_summary,
+                                           is_gce=is_gce,
+                                           email_subject_postfix=self.params.get('email_subject_postfix'))
+        except Exception as exc:  # noqa: BLE001
+            TestFrameworkEvent(
+                message='Failed to check regression',
+                source=self.__class__.__name__,
+                source_method='check_regression',
+                exception=exc
+            ).publish_or_dump()
 
     @staticmethod
-    def _calculate_average_latency(results):
+    def _calculate_average_max_latency(results):
         status = defaultdict(float).fromkeys(results[0].keys(), 0.0)
+        max_latency = defaultdict(list)
+
         for result in results:
             for key in status:
                 try:
                     status[key] += float(result.get(key, 0.0))
+                    if key in ["latency 95th percentile", "latency 99th percentile"]:
+                        max_latency[f"{key} max"].append(float(result.get(key, 0.0)))
                 except ValueError:
                     continue
 
         for key in status:
             status[key] = round(status[key] / len(results), 2)
 
+        for key, latency in max_latency.items():
+            status[key] = max(latency)
+
         return status
 
-    def warmup_cache(self, compaction_strategy):
-        cmd = f"cassandra-stress read no-warmup cl=QUORUM duration=180m -pop 'dist={self.get_cs_distribution()}' -mode native cql3 -rate 'threads=500 throttle=35000/s'"  # pylint: disable=line-too-long
-        stress_queue = self.run_stress_cassandra_thread(
-            stress_cmd=cmd,
-            stress_num=1,
-            compaction_strategy=compaction_strategy,
-            stats_aggregate_cmds=False,
-            round_robin=True
-        )
-        self.get_stress_results(stress_queue, store_results=False)
+    def warmup_cache(self, stress_cmd_templ):
+        stress_queue = []
+        for stress_cmd in stress_cmd_templ:
+            params = {"round_robin": True, "stats_aggregate_cmds": False}
+            stress_cmd_to_run = stress_cmd.replace("$threads", "1000")
```

Comment:
This is not needed.
We already picked the optimized number of threads to use in order to reach max throughput, so why shall we replace it with something that overloads the cluster?

Reply:
@fruch wants to get rid of this function. He suggests to add warm-up commands to the test yaml.
@roydahan and @fruch please, let me know what do you want to do with it

Reply:
I support that as well.

Reply:
I will handle this in the separate PR